examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,337 @@
+      metricValue: '$.duration',
+      unit: Unit.MILLISECONDS,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterRequestSize', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Request Size (bytes)',
+      filterPattern: FilterPattern.literal('{$.requestSize >= 0}'),
+      metricValue: '$.requestSize',
+      unit: Unit.BYTES,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterResponseSize', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Response Size (bytes)',
+      filterPattern: FilterPattern.literal('{$.responseSize >= 0}'),
+      metricValue: '$.responseSize',
+      unit: Unit.BYTES,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterGrpcStatusCode', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'GRPC Status Code',
+      filterPattern: FilterPattern.literal('{ $.status >= 0 }'),
+      metricValue: '$.status',
+    });
+  }
+
+  addWidgetsToDashboard(dashboard: cdk.aws_cloudwatch.Dashboard) {
+    const latency = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'Duration (Latency)',
+          namespace: 'MomentoMetricsCDKExample',
+        }),
+      ],
+      title: 'Duration (Latency)',
+    });
+
+    const errorCodes = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'GRPC Status Code',
+          namespace: 'MomentoMetricsCDKExample',
+        }),
+      ],
+      title: 'GRPC Error Codes',
+      period: cdk.Duration.minutes(1),
+      leftYAxis: {
+        label: 'Error Code',
+        max: 16,
+      },
+      leftAnnotations: [
+        {
+          value: 4,
+          fill: Shading.ABOVE,
+          label: 'GRPC Error Codes >= 4',
+        },
+      ],
+    });
+
+    const messageSizes = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'Response Size (bytes)',
+          namespace: 'MomentoMetricsCDKExample',
+          label: 'Response Size (bytes)',
+        }),
+        new Metric({
+          metricName: 'Request Size (bytes)',
+          namespace: 'MomentoMetricsCDKExample',
+          label: 'Request Size (bytes)',
+        }),
+      ],
+      title: 'Request and Response Sizes in Bytes',
+      period: cdk.Duration.minutes(1),
+      statistic: Stats.AVERAGE,
+    });
+
+    // Add the first three widgets in the first row of the dashboard
+    dashboard.addWidgets(latency, errorCodes, messageSizes);
+
+    const transactionsPerSecond = new SingleValueWidget({
+      metrics: [
+        new Metric({
+          metricName: 'Response Size (bytes)',
+          namespace: 'MomentoMetricsCDKExample',
+          label: 'Count of responses per second',
+          statistic: Stats.SAMPLE_COUNT,
+        }),
+      ],
+      period: cdk.Duration.seconds(1),
+      height: 6,
+      sparkline: true,
+      title: 'Transactions per second',
+    });
+
+    const numberOfRequests = new GraphWidget({
```

Comment:
i'm curious what is the intended difference between this widget and the previous? They are both basically graphing requests per second and I'm seeing the same values on the two graphs in my dashboard.

I also don't completely understand why the values are showing up as the same, when on one you are graphing sample count over the last minute and one you are graphing sample count over the last second :) something fishy about that that i don't know the answer to. So we need to sort that out, but probably also only need one of the two of these.

Also... I'm sleeping 100ms between requests, so I would expect to see about 10tps * 2 (one get and one set) = 20 tps, but I'm seeing about 1.1k, which seems way too high, and I don't have an answer for that yet either :)

Reply:
when i look at the lightstep metrics for this I see about 9.5 tps for get and 9.5 tps for set, which is much closer to what i was expecting: https://app.lightstep.com/momento-preprod/dashboard/user-experience/TJXd7Dq4?time_window=minutes_60&chart_id=XYbgq5zQ

Reply:
I think maybe what is happening here is that the maximum resolution for metricfilter metrics is 1 minute, so even when you set the period to something < 1 minute, you still get the resolution of 1 minute. (we should google this and see if this is documented for cloudwatch metric filters.)

if that's the case, then setting the period to 1m and dividing the value by 60 via cloudwatch metricmath may be the way to go. 1100 / 60 = 18.3, which is approximately what I'm seeing in lightstep for get + set combined. (with my own sleep between requests set to 100ms.)